app/grandchallenge/emails/tasks.py

```diff
@@ -0,0 +1,65 @@
+from celery import shared_task
+from django.contrib.auth import get_user_model
+from django.contrib.sites.models import Site
+from django.core.mail import send_mass_mail
+from django.core.paginator import Paginator
+
+from config import settings
+from grandchallenge.algorithms.models import Algorithm
+from grandchallenge.challenges.models import Challenge
+from grandchallenge.emails.utils import SendActionChoices
+from grandchallenge.reader_studies.models import ReaderStudy
+from grandchallenge.subdomains.utils import reverse
+
+
+@shared_task(**settings.CELERY_TASK_DECORATOR_KWARGS["acks-late-micro-short"])
+def send_bulk_email(action, subject, body):
+    if action == SendActionChoices.MAILING_LIST:
+        receivers = [
+            user
+            for user in get_user_model().objects.filter(
+                user_profile__receive_newsletter=True
+            )
+        ]
+    elif action == SendActionChoices.STAFF:
+        receivers = [
+            user for user in get_user_model().objects.filter(is_staff=True)
+        ]
+    elif action == SendActionChoices.CHALLENGE_ADMINS:
+        receivers = [
+            user
+            for challenge in Challenge.objects.all()
+            for user in challenge.admins_group.user_set.all()
+        ]
+    elif action == SendActionChoices.READER_STUDY_EDITORS:
+        receivers = [
+            user
+            for rs in ReaderStudy.objects.all()
+            for user in rs.editors_group.user_set.all()
+        ]
+    elif action == SendActionChoices.ALGORITHM_EDITORS:
+        receivers = [
+            user
+            for algorithm in Algorithm.objects.all()
+            for user in algorithm.editors_group.user_set.all()
+        ]

```

Comment:
Maybe split this out to another function and add a test that it does the right stuff?

Reply:
Also, this will fetch all of the users in memory at once, for this to work in production you will need to use querysets that return this data that are then fetched by the paginator.